ipatests/test_webui/ui_driver.py

```diff
@@ -362,6 +362,19 @@ def wait_for_request(self, implicit=0.2, n=1, d=0):
             self.wait()
         self.wait(d)
 
+    def wait_while_working(self, widget, implicit=0.2):
+        """
+        Wait while working widget active
+        """
+
+        working_widget = self.find('.working-widget', By.CSS_SELECTOR, widget)
+
+        self.wait(implicit)
+        WebDriverWait(self.driver, self.request_timeout).until_not(
+            lambda d: working_widget.is_displayed()
+        )
+        self.wait(0.5)
```

Comment:
Isn't 0.5s too much? At that point we know that the working widget is gone and thus the parent widget should be re-rendered. To be sure for the re-rendering, small wait like 0.1 can be there but 0.5 seems as unnecessary slowing down of the tests. Or is there a reason for it which I miss?

To be consistent with `wait_for_request`, it can be changed  to `self.wait(d)` and have `d` in method signature.

Otherwise it looks good. Btw, it would be good if somebody else would also look at this PR. I'm not doing any testing or more proper validation.

Reply:
With 0.2s it is quite unstable. So I decide set more. It's no so much compared with original timeout 0.2s + 3s (includes AJAX request).
Also this method is used for similar widgets, so there no sense to add parameter until more complicated case comes.

Reply:
Ok then, makes sense.